### ISSUE TITLE
unexpected failure to collect elasticache clusters

### ISSUE THREAD

**Issue by byron-janrain** [dtan4/terraforming#107]
I just got terraforming working but it dies when applying template to redis elasticache resources.  S3 works fine, ec2 works fine.
```
$ terraforming ecc
(erb):9:in `block in apply_template': undefined method `port' for nil:NilClass (NoMethodError)
	from (erb):1:in `each'
	from (erb):1:in `apply_template'
	from /usr/local/Cellar/ruby/2.2.2/lib/ruby/2.2.0/erb.rb:863:in `eval'
	from /usr/local/Cellar/ruby/2.2.2/lib/ruby/2.2.0/erb.rb:863:in `result'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/lib/terraforming/util.rb:4:in `apply_template'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/lib/terraforming/resource/elasti_cache_cluster.rb:19:in `tf'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/lib/terraforming/resource/elasti_cache_cluster.rb:7:in `tf'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/lib/terraforming/cli.rb:132:in `execute'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/lib/terraforming/cli.rb:28:in `ecc'
	from /usr/local/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /usr/local/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /usr/local/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /usr/local/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /usr/local/lib/ruby/gems/2.2.0/gems/terraforming-0.1.3/bin/terraforming:5:in `<top (required)>'
	from /usr/local/bin/terraforming:23:in `load'
	from /usr/local/bin/terraforming:23:in `<main>'
```
While Memcached clusters have a configuration endpoint, Redis clusters do not.  Looks like the [template](https://github.com/dtan4/terraforming/blob/master/lib/terraforming/template/tf/elasti_cache_cluster.erb) needs to be make the config endpoint properties conditional but I don't know ERB to submit pull.

**Comment by byron-janrain** [comment]
I've confirmed that when I pull the redis clusters out of my test environment and only memcache clusters remain, i get the expected behavior:
```
$ terraforming ecc
resource "aws_elasticache_cluster" "byron-test-mem" {
    cluster_id           = "byron-test-mem"
    engine               = "memcached"
    engine_version       = "1.4.14"
    node_type            = "cache.m1.small"
    num_cache_nodes      = 1
    parameter_group_name = "default.memcached1.4"
    port                 = 11211
    security_group_names = ["default"]
}
```

**Comment by dtan4** [comment]
@byron-janrain 
I fixed this issue at #112 and released this as v0.1.6 :shipit: Please try it.
If there is something wrong, please reopen this issue (or send pull request :octocat: )

**Comment by byron-janrain** [comment]
Confirmed working now.  Thanks for your timely response!
